lib/real-device-management.js

```diff
@@ -74,7 +74,9 @@ async function installToRealDevice(device, app, bundleId, opts) {
 
   const {skipUninstall, strategy, timeout} = opts ?? {};
 
-  if (!skipUninstall && bundleId && (await device.isAppInstalled(bundleId))) {
+  // (await device.isAppInstalled(bundleId)) could return not accurate app installation data
+  // so here should not reply on the result.
+  if (!skipUninstall && bundleId) {
     log.debug(`Reset requested. Removing app with id '${bundleId}' from the device`);
     await device.remove(bundleId);
   }
```

Comment:
perhaps, we may still want to install, then analyze the error and retry

Reply:
Yea, still we observe the same error after calling uninstall command explicitly before install, the next would be retrial

Reply:
Sorry, I am not following. the sentence above mentions

>  `async remove`  was able to uninstall the app completely from the device even that case.

and then here

> still we observe the same error after calling uninstall command explicitly before install

I would like to keep the current logic if possible as reinstalling of an app is an expensive operation, which slows down session startup

Reply:
Oh, sorry. Please ignore "still we observe the same error after calling uninstall command explicitly before install" case since that was potentially not correct as ambiguous situation.

I'll update this pr title also for offload case to focus on this.

Btw, `skipUninstall` is considered by `isAppInstalled` in `checkAutInstallationState`, so the `(await device.isAppInstalled(bundleId)` removal itself is ok since `skipUninstall` includes the check. It will reduce ~0.2 sec extra time to check `isAppInstalled`